### PR TITLE
Lap4 se 130

### PR DESCRIPTION
**Pull request by Mich-Elle-Lo**
**Note on File Changes**
High Number of File Changes: There are numerous file changes in this PR because I was **several** commits behind the develop branch. This PR includes updates from develop along with the new features and improvements mentioned below.

**Description**
This pull request includes the following updates and improvements:

**Profile Page Styling:**
Applied new styling to the profile page to enhance the user interface.
**Note:** Further styling enhancements are planned and will be implemented in subsequent updates.

**Backend Data Connection:**
Integrated profile page components with backend data to display user information dynamically.

**Refactor API Calls:**
Refactored the getUserData API call to include error handling, ensuring robust data retrieval and proper error messaging.

**Create setDoc Function:**
Implemented a setDoc function allowing users to edit their profile information. This function supports updating user details on the backend. (Will use updateDoc function afterwards)

**Edit Modal Integration:**
Connected the edit modal to the backend, enabling users to update their profile information directly from the modal interface.

**To Do**
Continue refining the profile page styling in future commits.
Further testing and validation of the new features and API integrations.

### CODE REVIEW

File: src/pages/ProfilePage/components/EditProfileModal/EditProfileModal.scss [hunk 1]
```diff
@@ -23,15 +26,26 @@
...
+
+    &::-webkit-scrollbar {
+      width: 0;
+      height: 0;
+    }
+
+    // Hide scrollbar for other browsers
+    -ms-overflow-style: none; // IE and Edge
+    scrollbar-width: none; // Firefox
   }
```
Comment (Bita-Fotovvat) [inline src/pages/ProfilePage/components/EditProfileModal/EditProfileModal.scss]: I got this deprecation warning for this part of the file: "Sass's behavior for declarations that appear after nested
rules will be changing to match the behavior specified by CSS in an upcoming version. To keep the existing behavior, move the declaration above the nested rule. To opt into the new behavior, wrap the declaration in `& {}`."

To make sure we avoid any issues in the future, you can wrap declarations in '& {}'

&::-webkit-scrollbar {
  width: 0;
  height: 0;
}
& {
  -ms-overflow-style: none; // IE and Edge
  scrollbar-width: none; // Firefox
}

or simply move the declarations above the nested rule.

File: src/pages/ProfilePage/components/EditProfileModal/EditProfileModal.tsx
Comment (Bita-Fotovvat) [inline src/pages/ProfilePage/components/EditProfileModal/EditProfileModal.tsx:1]: The edit functionality works for all the fields on basic info on profile details except for the discipline. Make sure to fix this part. I know that the functionality is yet to be completed on the next ticket, but just to make sure we know how it works as of now, for the skills part, I can remove the existing skills, but when I add a new item, it doesn't show it yet. And also the same for the portfolio.

**Review by Bita-Fotovvat — APPROVED** [review]
I ran the code, and it works perfect up to the point you've added to its functionality. Of course as you mentioned the functionality and styling is yet to be completed. Great job on the profile!
The profile page components are synced with backend data and shows the profile info.
The edit functionality works for all the fields on basic info on profile details except for the discipline, and the edit functionality on other parts will also be added.

File: src/pages/ProfilePage/components/EditProfileModal/EditProfileModal.tsx
Comment (Mich-Elle-Lo) [reply src/pages/ProfilePage/components/EditProfileModal/EditProfileModal.tsx:1]: It should work to add skill, have to press the + icon next to the newly added skill.

Will fix the nesting issues.

Comment (Bita-Fotovvat) [reply src/pages/ProfilePage/components/EditProfileModal/EditProfileModal.tsx:1]: I tested it and it works after I press the + icon.

**Pull request merged**